community/kernel/src/test/java/org/neo4j/kernel/api/index/SimpleIndexPopulatorCompatibility.java

```diff
@@ -149,6 +159,112 @@ public void shouldApplyUpdatesIdempotently() throws Exception
         }
     }
 
+    private List<NodeAndValue> allValues( Iterable<Value> supportedValues )
+    {
+        long nodeIds = 0;
+        List<NodeAndValue> result = new ArrayList<>();
+        for ( Value value : supportedValues )
+        {
+            result.add( new NodeAndValue( nodeIds++, value ) );
+        }
+        return result;
+    }
+
+    private List<NodeAndValue> allValues = allValues( testSuite.getSupportedValues() );
+
+    static class NodeAndValue
```

Comment:
Smells a bit like `IndexEntryUpdate`

Reply:
It does! We are amending this code quite a bit in the next PR, but maybe we change to only work with `IndexEntryUpdate` when we start homogenizing the test value generation as mentioned above.